scan-payment/src/main/java/com/getbouncer/scan/payment/analyzer/PaymentCardOcrAnalyzer.kt

```diff
@@ -0,0 +1,67 @@
+package com.getbouncer.scan.payment.analyzer
+
+import android.util.Log
+import com.getbouncer.scan.framework.Analyzer
+import com.getbouncer.scan.framework.AnalyzerFactory
+import com.getbouncer.scan.payment.ml.*
+import com.getbouncer.scan.payment.ml.ssd.DetectionBox
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+
+
+data class PaymentCardPanOcrAnalyzerOutput(
+    val pan: String?,
+    val detectedBoxes: List<DetectionBox>?,
+    val name: String?,
+    val expiry: String?,
+    val objBoxes: List<DetectionBox>?
+)
+
+class PaymentCardOcrAnalyzer private constructor(
+    private val coroutineScope: CoroutineScope,
+    private val ssdOcr: SSDOcr?,
+    private val nameDetect: NameDetectAnalyzer?
+) : Analyzer<SSDOcr.SSDOcrInput, PaymentCardOcrState, PaymentCardPanOcrAnalyzerOutput> {
+
+    override val name: String = "payment_card_ocr_analyzer"
+
+    override suspend fun analyze(data: SSDOcr.SSDOcrInput, state: PaymentCardOcrState): PaymentCardPanOcrAnalyzerOutput {
+        val cardDetectFuture = if (state.runNameExtraction && nameDetect != null) {
+            coroutineScope.async {
+                nameDetect.analyze(data, state)
+            }
+        } else {
+            null
+        }
+
+        val ocrFuture = if (state.runOcr && ssdOcr != null) {
+            coroutineScope.async {
+                ssdOcr.analyze(data, Unit)
+            }
+        } else {
+            null
+        }
+
+        return PaymentCardPanOcrAnalyzerOutput(
```

Comment:
if the containing method is a supervisor scope, we can omit the `return` key word

```suggestion
       PaymentCardPanOcrAnalyzerOutput(
```